RevenueCat/Scripts/Purchases.cs

```diff
@@ -88,7 +101,16 @@ private void Start()
     // Call this if you want to reset with a new user id
     private void Setup(string newUserId)
     {
-        _wrapper.Setup(gameObject.name, revenueCatAPIKey, newUserId, observerMode, userDefaultsSuiteName);
+        var apiKey = deprecatedLegacyRevenueCatAPIKey;
+        if (String.IsNullOrEmpty(apiKey))
+        {
+            if (Application.platform == RuntimePlatform.IPhonePlayer)
+                apiKey = revenueCatAPIKeyIOS;
+            else if (Application.platform == RuntimePlatform.Android)
+                apiKey = revenueCatAPIKeyAndroid;
```

Comment:
When adding Amazon we can check if `useAmazon` is true here and use its API Key.